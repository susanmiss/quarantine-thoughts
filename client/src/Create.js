import React, {useState} from 'react';
import axios from 'axios';
import Nav from './Nav';
import ReactQuill from 'react-quill'; 
import { getUser, getToken } from './helpers';
import 'react-quill/dist/quill.bubble.css';

const Create = () => {
    const [state, setState] = useState({
        title: '',
        user: getUser()
    })

    const [content, setContent] = useState('')

  const handleContent = (event) => {
      console.log(event) 
      setContent(event);
  }


    const {title, user} = state 
   
    const handleChange = (name) => (event) => {
        setState({...state, [name]: event.target.value})
    }

    const handleSubmit = event => {
        event.preventDefault() 
        axios.post(`${process.env.REACT_APP_API}/post`, {title, content, user}, {
            headers: {
                Authorization: `Bearer ${getToken()}` 
            }
        })
        .then(response => {
            console.log(response)
            setState({...state, title: '', user: ''});
            setContent('');
            alert(`Post titled ${response.data.title} is created.`)
        })
        .catch(error => {
            console.log(error.response)
            alert(error.response.data.error);
        })
    }    


    return (
        <div className="container p-5">
    
          <br />
          <h1>CREATE POST</h1>
          <br />
          {/* {JSON.stringify(state)} */}

          <form onSubmit={handleSubmit}>

              <div className="form-group" >
                  <label className="text-muted">Title</label>
                  <input onChange={handleChange('title')} value={title} type="text" className="form-control" placeholder="Post Title" required />
              </div>

              <div className="form-group">
                    <label className="text-muted">Content</label>
                    <ReactQuill
                        onChange={handleContent}
                        value={content}
                        theme="bubble"
                        className="pb-5 mb-3"
                        placeholder="Write something.."
                        style={{ border: '1px solid #666' }}
                    />
                </div>
      
              <div className="form-group" >
                  <label className="text-muted">User</label>
                  <input onChange={handleChange('user')} value={user} type="text" className="form-control" placeholder="Your Name Title" required />
              </div>
      
              <div>
                  <button className="primary">Send</button>
              </div>

          </form>
        </div>
      )  
}

export default Create;