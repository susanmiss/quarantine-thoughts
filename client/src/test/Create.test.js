import React from 'react';
import ReactDOM from 'react-dom'
import {shallow} from 'enzyme'; 
import renderer from 'react-test-renderer';
import Create from '../Create'


const setup = (props={}, state=null) => {
    const wrapper = shallow(<Create {...props} />)
    if( state ) wrapper.setState(state);
    return wrapper;
  }

  const findByTestAttr = (wrapper, val) => {
    return wrapper.find(`[data-test="${val}"]`);
  }

describe("<Create />", () => {

  let wrapper;
  beforeEach(() => wrapper = shallow(<Create />));
  
  it('Renders correctly', () => {
      expect(wrapper).toMatchSnapshot();
  })
  
  it('should render a <div >', () => { 
      expect(wrapper.find('div').length).toEqual(5);
  });

  it('should render a <form >', () => { 
    expect(wrapper.find('form').length).toEqual(1);
  });

  it('should render a <button>', () => { 
     expect(wrapper.find('button').length).toEqual(1);
  });


})