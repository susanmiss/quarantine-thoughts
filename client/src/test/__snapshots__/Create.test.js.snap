// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Create /> Renders correctly 1`] = `ShallowWrapper {}`;
